File: packages/thermal-state/android/src/main/java/io/capawesome/capacitorjs/plugins/thermalstate/classes/events/ThermalStateChangeEvent.java
package io.capawesome.capacitorjs.plugins.thermalstate.classes.events;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.thermalstate.interfaces.Result;

public class ThermalStateChangeEvent implements Result {

    @NonNull
    private final String state;

    public ThermalStateChangeEvent(@NonNull String state) {
        this.state = state;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("state", state);
        return result;
    }
}

File: packages/thermal-state/ios/Plugin/Classes/Events/ThermalStateChangeEvent.swift
import Foundation
import Capacitor

@objc public class ThermalStateChangeEvent: NSObject, Result {
    let state: String

    init(state: String) {
        self.state = state
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["state"] = state
        return result as AnyObject
    }
}
